core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1636,6 +1637,10 @@ private void restoreFile(final BlobStoreIndexShardSnapshot.FileInfo fileInfo, fi
                 } else {
                     stream = new RateLimitingInputStream(partSliceStream, restoreRateLimiter, restoreRateLimitingTimeInNanos::inc);
                 }
+
```

Comment:
> mostly bogus.

This can happen if we restore a shard and some files have the same name but different checksums or sizes. This can happen, when you try to restore a snapshot for a "parallel" future. For example, you create a snapshot `A`, then add few segments then make snapshot `B`, then restore `A`, index a few things and then try to restore `B`. `B` might contain some files with the same names but different content.